src/Core/src/Platform/iOS/AcceleratorExtensions.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Linq;
+using Foundation;
+using ObjCRuntime;
+using UIKit;
+
+namespace Microsoft.Maui.Platform
+{
+	internal static class AcceleratorExtensions
+	{
+		internal static UIMenuElement CreateMenuItem(this IMenuFlyoutItem virtualView, IMauiContext mauiContext)
+		{
+			int index = MenuFlyoutItemHandler.menus.Count;
+			UIImage? uiImage = virtualView.Source.GetPlatformMenuImage(mauiContext);
+			var selector = new Selector($"MenuItem{index}:");
+
+			bool selectorFound =
+				MauiUIApplicationDelegate.Current.RespondsToSelector(selector);
+
+			if (!selectorFound)
+			{
+				throw new InvalidOperationException(
+					$"By default we only support 50 MenuItems. You can add more by adding the following code to {MauiUIApplicationDelegate.Current.GetType()}\n\n" +
```

Comment:
> was there no other way we can do menus? (not for this PR, but in some future PR)

I mean, if I found the other way, I would have done it that way :-p I poked Soto about this a little bit and this is where we landed. 

From what I remember every menu item needs a unique selector, but we can't generate selectors at runtime. We could swizzle them, but nobody will let me swizzle anything :-(  Or I'm wrong which would be cool.

Somewhere on this PR there's some additional conversation around this because the plan was to switch these all to `UIAction` which would then allow for infinite menus, but it looks like `UIAction` doesn't support accelerators. So, after this will come a PR where the limitation will only be hit if you create 50 menu items with accelerators. 

Unless during the creation of that next PR someone finds a better solution.

Reply:
Yeah, I had some hopes...